### PR TITLE
Flexbe update

### PR DESCRIPTION
**Pull request by dcconner**

### CODE REVIEW

File: .docker/entrypoint.sh [hunk 1]
```diff
@@ -56,6 +56,11 @@ function delib_clean () {
         cd $cwd
 }
 
+function qt_rendering() {
```
Comment (sea-bass) [inline .docker/entrypoint.sh:59]: consider naming this `qt_software_rendering` or something that indicates it's switching to software rendering.

File: technologies/FlexBE/README.md [hunk 2]
```diff
@@ -55,10 +60,68 @@ To run start the following in separate terminals
...
+* `test pick place` - Navigation with pick and place using behavior parameters
+* `DetectSelect` - Move, detect objects, select using `input_action_server`, move and place
+* `Go Beh` - Go to target location
+
+Use `clear; ros2 run delib_ws_worlds run --ros-args -p problem_number:=1`
+
+* `delib_ws_p2` - use behavior parameters plan to door, open door, then use `DetectSelect` embedded behavior
+* `delib_ws_p2_sm` - use hierarchical state machine to select door, open, then use `DetectSelect` embedded behavior
+* `Through Door` - Travel through doorway opening if necessary
+* `Traverse` - Go to specified target location, opening intermediate doors if necessary using `Go Beh` and `Through Door`
```
Comment (sea-bass) [inline technologies/FlexBE/README.md:115]: is `Go Beh` correct, or a typo?

(I checked, and I guess it's correct -- maybe consider expanding out the name? Is it supposed to mean "behavior"?)

File: technologies/FlexBE/pyrobosim_flexbe_utilities/pyrobosim_flexbe_utilities/world_configuration.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+    # Find the nearest point on the line segment
+    nearest = line_start + projection * line_unitvec
+
+    # Return the Euclidean distance between the point and the nearest point on the line segment
+    return np.linalg.norm(nearest - point)
+
+
+# Helper function:
+def is_point_inside_polygon(point, polygon):
```
Comment (sea-bass) [inline technologies/FlexBE/pyrobosim_flexbe_utilities/pyrobosim_flexbe_utilities/world_configuration.py:59]: This works, but if you want, you could consider using the `shapely` package which I use internally in PyRoboSim to do these calculations.

**Review by sea-bass — APPROVED** [review]
Tested the `PatrolCharge` behavior and, barring the UI sync issue being discussed, works great! The robot has been retrying/recharging with no problems.

Just left a few super minor comments, but approving.

File: problems/delib_ws_worlds/worlds/world4.yaml [hunk 4]
```diff
@@ -206,7 +206,7 @@ locations:
       parent: kitchen
       category: storage
       pose: [-3.65, -1.5, 0.0, 1.57]
-      is_open: true
+      is_open: false
```
[the remark this answers is not in the document]
Comment (dcconner) [reply problems/delib_ws_worlds/worlds/world4.yaml:209]: the problem description says that pantry is closed, but world has is_open true.  This is making consistent with the problem description on README.

File: .docker/entrypoint.sh (hunk 1 above)
Comment (dcconner) [reply .docker/entrypoint.sh:59]: will do.  how about `qt_soft_render` just to keep shorter ?

File: technologies/FlexBE/README.md (hunk 2 above)
Comment (dcconner) [reply technologies/FlexBE/README.md:115]: Was short for "Go Behavior"   I had some not well thought out names during testing that I should clean up, but I wanted to get something up.

File: .docker/entrypoint.sh (hunk 1 above)
Comment (sea-bass) [reply .docker/entrypoint.sh:59]: Sure!

**Pull request merged**